common/v2/services/Store/BalanceService.tsx

```diff
@@ -82,12 +93,14 @@ const getTokenBalances = (
   address: TAddress,
   tokens: StoreAsset[]
 ): BalanceMap => {
-  return tokens.reduce(async (balances, token) => {
-    return {
-      ...balances,
-      [token.contractAddress as TAddress]: await provider.getRawTokenBalance(address, token)
-    };
-  }, {});
+  return toBigNumberJS(
+    tokens.reduce(async (balances, token) => {
+      return {
+        ...balances,
+        [token.contractAddress as TAddress]: await provider.getRawTokenBalance(address, token)
+      };
+    }, {})
+  );
```

Comment:
The `Promise` here is ignored, so this would throw an error if `provider.getRawTokenBalance` fails. You can solve this by reducing a `Promise` and returning a `Promise<BalanceMap>` from this function instead, for example:

https://github.com/MyCryptoHQ/eth-scan/blob/master/src/utils/batch.ts#L34-L36

Reply:
Could you verify my changes?